service/volume.py

```diff
@@ -67,9 +103,47 @@ def create_volume(esh_driver, identity_uuid, name, size,
         metadata=metadata,
         snapshot=snapshot,
         image=image)
+
+    if not success and raise_exception:
+        raise exceptions.VolumeError("The volume failed to be created.")
```

Comment:
I could see `VolumeError` being used for actual errors and something like `NotFound` or a derivative of `ServiceException` being more of a warning or bad request.